src/Analyzer.Utilities/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -124,7 +124,113 @@ public static bool HasFinalizer(this INamedTypeSymbol symbol)
             return symbol.GetMembers()
                 .Where(m => m.Kind == SymbolKind.Method)
                 .Cast<IMethodSymbol>()
-                .Any(IMethodSymbolExtensions.IsFinalizer);
+                .Any(m => m.IsFinalizer());
+        }
+
+        public static bool HasAttribute(this INamedTypeSymbol symbol, INamedTypeSymbol attribute)
```

Comment:
I added an XML doc comment on the method and documented the existing behavior, which is that it does not find attributes on base types.